File: 682.baseball-game.ts
/*
 * @lc app=leetcode id=682 lang=typescript
 *
 * [682] Baseball Game
 */

// @lc code=start
function calPoints(operations: string[]): number {
  let newArr: number[] = [];
  for (let i = 0; i < operations.length; i++) {
    switch (operations[i]) {
      case "+":
        newArr.push(newArr[newArr.length - 1] + newArr[newArr.length - 2]);
        break;
      case "D":
        newArr.push(newArr[newArr.length - 1] * 2);
        break;
      case "C":
        newArr.pop();
        break;
      default:
        newArr.push(parseInt(operations[i]));
        break;
    }
  }
  return newArr.reduce((sum, num) => sum + num, 0);
}
// @lc code=end

File: 682.baseball-game.py
#
# @lc app=leetcode id=682 lang=python3
#
# [682] Baseball Game
#

# @lc code=start
from typing import List


class Solution:
    def calPoints(self, operations: List[str]) -> int:
        ans = []
        j = 0
        for i in range(len(operations)):
            if operations[i] == '+':
                ans.append(int(ans[j-1]) + int(ans[j-2]))
                j += 1
            elif operations[i] == 'D':
                ans.append(int(ans[j-1]*2))
                j += 1
            elif operations[i] == 'C':
                ans.pop()
                j -= 1
            else:
                ans.append(int(operations[i]))
                j += 1
        return sum(ans)

# @lc code=end
